src/sign_tx.c

```diff
@@ -0,0 +1,449 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <os.h>
+#include <os_io_seproxyhal.h>
+#include <string.h>
+#include "hathor.h"
+#include "util.h"
+#include "ux.h"
+
+static sign_tx_context_t *ctx = &global.sign_tx_context;
+
```

Comment:
Can you include a description of how it works, please? What confirmations are expected from the user, what "screens" are shown, etc?

Reply:
Added a description to the top of the file. Can you check if it helps?

Reply:
Sure. Thanks!

Reply:
The description is great. It will be way easier to review now. Thanks again!